framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2919,20 +2936,10 @@ public Void visitIdentifier(IdentifierTree node, Void p) {
     }
 
     protected void checkAccess(IdentifierTree node, Void p) {
-        // Do not add functionality to this method body.
-        // Add it in the method called below instead.
-        checkAccess(node, p, true);
-    }
-
-    protected void checkAccess(IdentifierTree node, Void p, boolean doCheckPreconditions) {
         MemberSelectTree memberSel = enclosingMemberSelect();
         ExpressionTree tree;
         Element elem;
 
-        if (doCheckPreconditions) {
-            checkPreconditions(node, TreeUtils.elementFromUse(node), false);
```

Comment:
Yes, I am removing changes I made a long time ago to support adding a PreconditionAnnotation to the GuardedBy type qualifier. This strategy can work for @GuardedBy by name, but not for @GuardedBy by value. Also, if we do use this strategy again for @GuardedBy by name, BaseTypeVisitor is not the right place to do it - it should be localized to the Lock Checker.